src/gengowatcher/watcher.py

```
@@ -1135,44 +1231,46 @@ def _process_new_job(self, job_id, title, reward, url, source, source_meta=None)
+            job_added = False
+            try:
+                inserted = self.state.add_job(job_data)
+                if inserted:
+                    job_added = True
+                    # Only update bookkeeping after successful add_job
+                    self._seen_jobs_session.add(job_id)
+                    self.state.seen_job_ids.append(job_id)
+                    self.state.total_new_entries_found += 1
+                    self.session_new_entries += 1
+                    self.session_total_value += reward
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)